### DISCUSSION TITLE
Binding select_single to functions

### DISCUSSION THREAD

**Discussion by erlebach** [vega/altair#2463] [Q&A] [answered]
I would like to create a dropdown menu with the names of aggregation functions (mean, median, max), and be able to select them via clicking, and apply this function to the data on the y axis in a mark_bar chart. Is this possible? I have not found a single example of anything similar. 

Thanks,   Gordon.

**Comment by jakevdp** [comment]
You can't bind a select to a function, but you can use a calculate transform to pre-compute values that you then bind to a select statement. Does that make sense?

**Comment by SeeBastion156** [comment]
I have the same problem, I want to the user to use a dropdown menu to select the aggregation function and can't get it run. Is there a small example how to get this work?

**Reply by joelostblom** [reply, marked as answer]
Combining from [this](https://altair-viz.github.io/user_guide/transform/aggregate.html) and [this](https://altair-viz.github.io/user_guide/interactions.html#encoding-channel-binding) docpage, you could do something like:

```py
from vega_datasets import data
import altair as alt

dropdown = alt.binding_select(
    options=['median_acc', 'mean_acc'],
    name='X-axis column '
)
xcol_param = alt.param(
    value='mean_acc',
    bind=dropdown
)
cars = data.cars()
alt.Chart(cars, title=alt.Title(alt.expr(xcol_param.name))).mark_bar().encode(
    y='Cylinders:O',
    x=alt.X('selected_column:Q').title(None)
).transform_aggregate(
    mean_acc='mean(Acceleration)',
    median_acc='median(Acceleration)',
    groupby=["Cylinders"]
).transform_calculate(
    selected_column=f'datum[{xcol_param.name}]'
).add_params(
    xcol_param
)
```

**Reply by SeeBastion156** [reply]
Thx for the example. I copied your code for testing, but it will not work. I got error messages:

Error: Unrecognized signal name: "param_31"

**Reply by joelostblom** [reply]
That's odd. Could you try to restart your IDE and kernel and make sure you're on altair >= 5?

**Comment by SeeBastion156** [comment]
I restarted IDE and kernel. I use the latest altair version 5.3.0. I have to say that I push the plot to streamlit. I use python 3.10.13. No luck.

**Reply by mattijn** [reply]
Can you try it without streamlit first?
Maybe this issue at the streamlit repo is related: https://github.com/streamlit/streamlit/issues/8765?

In https://github.com/vega/altair/issues/3416 was some discussion in relation to streamlit and altair selection mechanism.

**Reply by ChristopherDavisUCI** [reply]
I apologize if my comment completely misses the point, but would it work in your use case to define the dropdown menu on the Streamlit side rather than on the Altair side?

**Reply by SeeBastion156** [reply]
> I apologize if my comment completely misses the point, but would it work in your use case to define the dropdown menu on the Streamlit side rather than on the Altair side?

This is currently my fallback variant, but much less elegant.
